src/platform/plugins/shared/share/public/components/export_popover.tsx

```diff
@@ -0,0 +1,313 @@
+                isDisabled={menuItem.config.disabled}
+                onClick={() => {
+                  setSelectedMenuItemId(menuItem.id);
+                  setIsFlyoutVisible(true);
+                }}
+              />
+            </EuiToolTip>
+          ))}
+        </EuiListGroup>
+      </EuiWrappingPopover>
+      {isFlyoutVisible && (
+        <EuiFlyout
+          data-test-subj="exportItemDetailsFlyout"
+          size="s"
+          onClose={flyoutOnCloseHandler}
+          css={() => ({
+            ['--euiFixedHeadersOffset']: 0,
+          })}
+          ownFocus
+          maskProps={{
+            headerZindexLocation: 'above',
+          }}
+        >
+          <EuiFlyoutHeader hasBorder>
+            <EuiTitle>
+              <h2>
+                <FormattedMessage
+                  id="share.export.flyoutTitle"
+                  defaultMessage="Export {objectType} as {type}"
+                  values={{ objectType, type: selectedMenuItem?.config.label }}
+                />
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+            <EuiFlexGroup direction="column">
+              <Fragment>
+                {selectedMenuItem?.config.renderLayoutOptionSwitch && (
+                  <EuiFlexItem>
+                    <LayoutOptionsSwitch
+                      usePrintLayout={usePrintLayout}
+                      printLayoutChange={(evt) => setPrintLayout(evt.target.checked)}
+                    />
+                  </EuiFlexItem>
+                )}
+              </Fragment>
+              <Fragment>
+                {selectedMenuItem?.config.renderCopyURIButton && publicAPIEnabled && (
+                  <EuiFlexItem>
+                    <EuiFormRow
+                      label={
+                        <EuiText size="s">
+                          <h4>
+                            <FormattedMessage
+                              id="share.export.postURLHeading"
+                              defaultMessage="Post URL"
+                            />
+                          </h4>
+                        </EuiText>
+                      }
+                      fullWidth
+                    >
+                      <EuiFlexGroup direction="column">
+                        <EuiFlexItem>
+                          <EuiText size="s" color="subdued">
+                            <FormattedMessage
+                              id="share.export.postURLDescription"
+                              defaultMessage="Allows to generate selected file format programmatically outside Kibana or in Watcher."
+                            />
+                          </EuiText>
+                        </EuiFlexItem>
+                        <EuiFlexItem>
+                          <EuiCodeBlock
+                            data-test-subj="exportAssetValue"
+                            css={{ overflowWrap: 'break-word' }}
+                            language="text"
```

Comment:
It would be nice if we could specify our own props to the code block, or maybe at least the language. For the Dashboard as JSON export, we would want to set `language="json"`.